AUT-866

vaitkusd opened the pull request:
feat: Full atom mapping module added, including main functions, example notebook, and unit tests.

vaitkusd committed:
AUT-866

Full atom mapping module added, including main functions, example notebook, and unit tests.

feat: INCA_input_parser() now returns metabolite annotations.

vaitkusd committed:
tests: adjusted unit tests

vaitkusd committed:
fix: adaptation to run in Windows

Added platform and subprocess modules to help correctly run different shell commands in all OS's

vaitkusd committed:
tests: fix item sorting

Sorted os.listdir output alphabetically to standardize file reading order

vaitkusd committed:
tests: update MFA_visualization tests

Changed assertEqual to assertAlmostEqual in test_reshape_fluxes_escher()

mmattano commented on BFAIR/atom_mapping/atom_mapping.py:
```diff
@@ -0,0 +1,697 @@
+        
+        Parameters
+        ----------
+        metabolite_data: pandas.DataFrame
```
Could you please add a space between the variable names and the colons for the object type descriptions? It's numpy docstring thing (section 4 here https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)

vaitkusd committed:
tests: added AlmostEqual dict assertion

New method added to compare whether two dictionaries are almost equal

vaitkusd committed:
fix: fixed docstring formatting

vaitkusd committed:
tests: added method for list AlmostEqual assertion

Issue: Different OS's generate -0 instead of 0 in some molecule coordinates.

Solution: converted all numeric values in lists to floats instead of strings, and used new method to assertAlmostEqual on lists.

vaitkusd committed:
fix: standardized formatting

vaitkusd committed:
fix: last reformat to please flake8

vaitkusd committed:
fix: sorted file reading order

vaitkusd committed:
build: updated config.yml to install Java
